plugins/inputs/sqlserver/azuresqldbqueries.go

```diff
@@ -233,13 +233,27 @@ SELECT
 		WHEN slo.[edition] = 'Hyperscale' then NULL
 		ELSE CAST(DATABASEPROPERTYEX(DB_NAME(),'MaxSizeInBytes') as bigint)/(1024*1024)
 	END AS [total_storage_mb]
+	,CASE
+		WHEN slo.[edition] = 'Hyperscale' then NULL
+		ELSE (SELECT SUM(CAST(FILEPROPERTY(name, 'SpaceUsed') AS INT) / 128) FROM sys.database_files WHERE type_desc = 'ROWS')
+	 END AS used_space_data_mb
+	 ,CASE
+		WHEN slo.[edition] = 'Hyperscale' then NULL
+		ELSE (SELECT total_log_size_in_bytes / (1024 * 1024) FROM sys.dm_db_log_space_usage)
+	 END AS total_log_mb
+	 ,CASE
+		WHEN slo.[edition] = 'Hyperscale' then NULL
+		ELSE (SELECT used_log_space_in_bytes / (1024 * 1024) FROM sys.dm_db_log_space_usage)
+	 END AS used_space_log_mb
+	/*
 	,CASE
 		WHEN slo.[edition] = 'Hyperscale' then NULL
 		ELSE (
 			cast(DATABASEPROPERTYEX(DB_NAME(),'MaxSizeInBytes') as bigint)/(1024*1024) -
 			(select SUM([size]/128 - CAST(FILEPROPERTY(name, 'SpaceUsed') AS int)/128) FROM sys.database_files)
 		)
 	END AS [available_storage_mb]
+	*/
```

Comment:
Commenting this part means `available_storage_mb` will not be in the resulting metrics anymore? This looks like a breaking change for current users?

Reply:
You are right and this is why I left the original code commented. 

From my point of view, it will introduce a breaking change but current available_storage_mb computation is not correct. From my point of view, we have different options:

1) To remove available_storage_mb column and leave the user to compute it with other new available columns
2) To leave available_storage_mb but we have to change the formula because it will introduce some confusion otherwise (my 2 cents)

I'm in favor of the solution 1) but what are you thoughts ?

Reply:
Better to remove it instead of commenting, it will be more clear for the reviewers the part is not in use anymore.

Hmm, both of the solutions are breaking for current users. What do you think @MyaLongmire @srebhan ?

Reply:
I use telegraf and this plugin on thousands of instances and I think with whatever decision we make we should call it out in a changelog in the readme (for this or any other potentially impactful fixes).  

I noticed recently a telegraf.exe upgrade adding backup related queries that we didn't have in our org's telegraf.conf exclusions.  I'm not upset by it but if we put those things at the top of the readme and we let the community know to check the readme before upgrading to look for any new features or breaking changes I think the experience would be smoother.  

I also think a lot of us might be using similar "SQL Managed Instance" Grafana Dashboards and if the readme linked to a copy of them we might be able to move organizations dashboards and plugins in lockstep to really help out. 

IMHO if its possible to fix the formula rather than removing the column it would be better than a complete removal.  If we want to remove it completely I think we definitely need to call it out in the readme and it may be more palatable if we document it in the readme and let it soak for a release

I like the work you're doing @dba-leshop please keep it up!

Reply:
@dba-leshop I think we can go with option 2; keep the field but fix the formula.

Reply:
> You are right and this is why I left the original code commented for the moment.
> 
> It will introduce a breaking change for sure but current available_storage_mb computation is not correct. From my point of view, we have different options:
> 
> 1. To remove available_storage_mb column and leave the user to compute it with other new available columns
> 2. To leave available_storage_mb but we have to change the formula because it will introduce some confusion otherwise (my 2 cents)
> 
> I'm in favor of the solution 1) but what are you thoughts ?

@denzilribeiro @masree can we have your opinion on this one?

Reply:
Awaiting feedback from @denzilribeiro  @masree to implement the final solution and fix the conflict. Option 2 seems to be preferred one so far.

Reply:
Since we still have no feedback, I'd vote for option 2, (which doesn't break compatibility)
_To leave available_storage_mb but we have to change the formula because it will introduce some confusion otherwise (my 2 cents)_

Reply:
Leaving the field sounds preferable to avoid breaking users. Be sure to include a comment in the README about the change in how it is computed.

Reply:
I agree that option 2 is preferred to avoid a breaking change. Just make sure to keep the special case for Hyperscale in the definition of `available_storage_mb` because this metric is undefined in that case.